cpp/src/strings/datetime.cu

```diff
@@ -0,0 +1,763 @@
+        thrust::transform( rmm::exec_policy(stream)->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(results_view.size()),
+            d_results, pfn );
+    }
+    template <typename T, std::enable_if_t<not cudf::is_timestamp<T>()>* = nullptr>
+    void operator()( column_device_view const&,
+                     std::string const&, timestamp_units,
+                     mutable_column_view&,
+                     cudaStream_t ) const
+    {
+        CUDF_FAIL("Only timestamps type are expected");
+    }
+};
+
+} // namespace
+
+//
+std::unique_ptr<cudf::column> to_timestamps( strings_column_view const& strings,
+                                             data_type timestamp_type,
+                                             std::string const& format,
+                                             rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                             cudaStream_t stream = 0 )
+{
+    size_type strings_count = strings.size();
+    if( strings_count==0 )
+        return make_timestamp_column( timestamp_type, 0 );
+
+    CUDF_EXPECTS( !format.empty(), "Format parameter must not be empty.");
+    timestamp_units units = cudf::experimental::type_dispatcher( timestamp_type, dispatch_timestamp_to_units_fn() );
+
+    auto strings_column = column_device_view::create(strings.parent(), stream);
+    auto d_column = *strings_column;
+
+    auto results = make_timestamp_column( timestamp_type, strings_count,
+        copy_bitmask( strings.parent(), stream, mr), strings.null_count(), stream, mr);
+    auto results_view = results->mutable_view();
+    cudf::experimental::type_dispatcher( timestamp_type, dispatch_to_timestamps_fn(),
+                                         d_column, format, units,
+                                         results_view,
+                                         stream );
+    results->set_null_count(strings.null_count());
+    return results;
+}
+
+} // namespace detail
+
+// external API
+
+std::unique_ptr<cudf::column> to_timestamps( strings_column_view const& strings,
+                                             data_type timestamp_type,
+                                             std::string const& format,
+                                             rmm::mr::device_memory_resource* mr )
+{
+    return detail::to_timestamps( strings, timestamp_type, format, mr );
+}
+
+namespace detail
+{
+namespace
+{
+// converts a timestamp into date-time string
+template <typename T>
+struct datetime_formatter
+{
+    const column_device_view d_timestamps;
+    const format_item* d_format_items;
+    size_type items_count;
+    timestamp_units units;
+    const int32_t* d_offsets;
+    char* d_chars;
+
+    // divide timestamp integer into time components (year, month, day, etc)
+    // TODO call the simt::std::chrono methods here instead when the are ready
```

Comment:
@trxcllnt are they ready?

Reply:
@trxcllnt ?

Reply:
These methods are in https://github.com/rapidsai/cudf/pull/3201, so yes it's ready. We can either wait till #3201 is merged, or switch the `libcudacxx` submodule reference in this PR first.

Reply:
@trxcllnt I'm not seeing the `parse` and `format` functions in the `include/simt/chrono` file in #3201   Can you verify that these are in the `libcudacxx` submodule we will be using?

Reply:
@davidwendt `libcudacxx` doesn't include parse and format yet afaik because those need `std::string`, but the other C++20 calendar types like `year_month_day` are there.

Reply:
@jrhemstad Do you want me to rewrite these APIs in this PR? I suppose we can close this PR and work on a rewrite in the next release.

Reply:
Repeating that the `simt::chrono` code in `libcudaxx` does **not** include `parse` and `format`.

Reply:
So what's the status of this? Is this a blocker @davidwendt ? I don't follow the full details.

Reply:
We are good to go on this. 
We were hoping the `simt::chrono` and `parse` and `format` functions were going to part of the `libcudacxx` but it does not look like they will be anytime soon.
There are some other `simt::chrono` functions we could re-use here in the future but these are not blocking this PR.